sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/Driver.java

```diff
@@ -0,0 +1,29 @@
+package com.mesosphere.sdk.scheduler;
+
+import org.apache.mesos.SchedulerDriver;
+
+import java.util.Optional;
+
+/**
+ * This class provides global access to a SchedulerDriver.
+ */
+public final class Driver {
+    private static SchedulerDriver driver;
+    private static Object lock = new Object();
+
+    private Driver() {
+        // Don't instantiate this class.
+    }
+
+    public static Optional<SchedulerDriver> getDriver() {
+        synchronized (lock) {
```

Comment:
Most of the `getDriver` calls have explicit `isPresent()` checks after the calls to log error messages when this is not the case. Would it make sense to just `throw` instead as this seems to indicate a critical bug on our side?

Reply:
I wouldn't do it here.  In the places where I've written, "This should never happen." it might make sense to throw.  The reason I chose not to, is that although it shouldn't ever happen it's also possible it will self resolve...  Maybe we should crash / throw though.  `IllegalStateException` might make sense.

Reply:
I'm adding throws on the client side.  That is I'm letting consumers decide whether the absence of a driver a particular point is a cause for concern, and how much concern.

Reply:
Sounds good. It wasn't clear when it was fatal, and when it could be tolerated which is why I asked.